src/MonoTorrent.Client/MonoTorrent/Torrent.cs

```diff
@@ -712,32 +760,45 @@ static PieceHashesV2 LoadHashesV2 (IList<ITorrentFile> files, BEncodedDictionary
             return new PieceHashesV2 (files, hashes);
         }
 
-        static void LoadTorrentFilesV2 (string key, BEncodedDictionary data, List<ITorrentFile> files, int pieceLength, ref int totalPieces, string path)
+        static void LoadTorrentFilesV2 (string key, BEncodedDictionary data, List<ITorrentFile> files, int pieceLength, ref int totalPieces, string path, bool isHybrid)
         {
             if (key == "") {
                 var length = ((BEncodedNumber) data["length"]).Number;
                 if (length == 0) {
-                    files.Add (new TorrentFile (path, length, 0, 0, 0));
+                    files.Add (new TorrentFile (path, length, 0, 0, 0, TorrentFileAttributes.None, 0));
                 } else {
                     totalPieces++;
-                    var offsetInTorrent = (files.LastOrDefault ()?.OffsetInTorrent ?? 0) + (files.LastOrDefault ()?.Length ?? 0);
+                    var offsetInTorrent = (files.LastOrDefault ()?.OffsetInTorrent ?? 0) + (files.LastOrDefault ()?.Length ?? 0) + (files.LastOrDefault ()?.Padding ?? 0);
                     var piecesRoot = data.TryGetValue ("pieces root", out var value) ? ((BEncodedString) value).AsMemory () : ReadOnlyMemory<byte>.Empty;
-                    files.Add (new TorrentFile (path, length, totalPieces, totalPieces + (int) (length / pieceLength), offsetInTorrent, piecesRoot));
+
+                    // TODO JMIK: wrong calculation for endIndex, when length is exactly pieceLength?
+                    files.Add (new TorrentFile (path,
```

Comment:
Is it feasible to rewrite this method so it creates a bunch of `TorrentFileTuple` (just like the nice cleanup you did for the v1 method) and then both the v1 and v2 implementation could rely on the logic in `TorrentFile.Create` to set the right start/end piece indices etc?